doc/design_decisions.rst

```diff
@@ -0,0 +1,87 @@
+Design decisions
+================
+
+
+Single Pool instance
+--------------------
+
+We have decided to keep all packages incl. comps in a single Pool instance.
+Lookups are slower due to iterating through a larger data set
+(using smaller dedicate Pool instances per package type would be faster),
+but we don't have to duplicate repo definitions, keep the repos in sync
+across all Pool instances etc.
+
+Modularity implementation is an exception to this rule
+because the modulemd records require special treatment.
+
+
+Working with multiple package types
+-----------------------------------
+
+It is possible to load multiple package types (such as rpm, deb, arch, ...)
+in a single libsolv Pool and make them available via PackageSack.
+We recommend doing so only for querying purposes.
+We MUST avoid running a transaction across multiple package types
+because the respective databases (rpmdb, dpkg database, ...) would contain
+partial records with broken dependencies and would be impossible to manage.
```

Comment:
Just a note, it feels to me this lacks a lot of details of e.g. the setup of the underlying system, like would the system have both rpm and dpkg managing parts of the packages? Or is that already ruled out by the "MUST avoid running a transaction across multiple package types" part?

This touches the subject of the two use cases "running dnf as the system package manager" and "running dnf to inspect arbitrary repositories (potentially with varying package types)". Have we ever reached a consensus on this? I'd still like to explicitly separate the use cases with specific parts of the API to cater to both. Mainly so that with the latter one (inspect arbitrary repos) the user can easily run dnf that's completely independent of the system it runs on (doesn't need to be root, has a separate cache) and just allows to query separately defined repos in an isolated way.

Reply:
I've rephrased the sentence a bit.
Yes, it's ruled out by the "MUST".

We probably need to discuss and document the suggested use cases (system package manager vs queries) in a different PR.